src/zcash/JoinSplit.cpp

```diff
@@ -303,6 +304,12 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
             }
 
             out_ephemeralKey = encryptor.get_epk();
+
+            // !!! Payment disclosure START
+            if (out_ephemeralSecretKey != nullptr) {
```

Comment:
this seems odd. did you mean `out_ephemeralKey`? So far `out_ephemeralSecretKey` is `nullptr` and hasen't  been set otherwise

Reply:
@arielgabizon So what's happening is that `prove(param, param,... , uint256 *out_ephemeralSecretKey)` is being invoked.  If out_ephemeralSecretKey (a pointer to a uint256) is not set, do nothing, as the caller passed in a nullptr because they don't want the esk.  If a pointer to an uint256 was actually passed in, then set the value of that uint256 to the esk.